### ISSUE TITLE
(OpenBSD 5.7) configure: error: Detected LibreSSL: This is NOT supported, and may break consensus compatibility!

### ISSUE THREAD

**Issue by ghost** [bitcoin/bitcoin#6486]

**Comment by luke-jr** [comment]
Yes, install OpenSSL instead.

**Comment by wolololo** [comment]
OpenSSL installed, but does not solve the problem.

**Comment by luke-jr** [comment]
Please post config.log and what version of OpenSSL you are using.

**Comment by wolololo** [comment]
225 KB (229.387 B)
http://vpaste.net/ni5IC

**Comment by wolololo** [comment]
I compile it --with-libressl flag.

**Comment by luke-jr** [comment]
The paste is incomplete; what version of OpenSSL is this?

**Comment by wolololo** [comment]
OpenSSL 1.0.1k

**Comment by laanwj** [comment]
The problem here, it appeared, is that OpenBSD, when you use `pkg_add openssl` installs openssl as the pkg-config package `eopenssl`:
```
orion@lethe:~/bitcoin$ pkg-config --libs --cflags openssl  
-I/usr/include -L/usr/lib -lssl -lcrypto
orion@lethe:~/bitcoin$ pkg-config --libs --cflags eopenssl 
-I/usr/local/include -L/usr/local/lib/eopenssl -lssl -lcrypto
```
But couldn't even work around this with the painful:
```
./configure ... SSL_CFLAGS="$(pkg-config --cflags eopenssl)" SSL_LIBS="$(pkg-config --libs eopenssl)" CRYPTO_CFLAGS="$(pkg-config --cflags eopenssl)" CRYPTO_LIBS="$(pkg-config --libs eopenssl)"
```
config.log shows:
```
configure:26667: checking for a supported OpenSSL version
configure:26682: eg++ -o conftest -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter  -Wstack-protector -fstack-protector-all -fPIE  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2   -Wl,-z,relro -Wl,-z,now -pie conftest.cpp  -L/usr/local/lib/eopenssl -lssl -lcrypto -L/usr/local/lib/eopenssl -lssl -lcrypto >&5
conftest.cpp: In function 'int main()':
conftest.cpp:68:14: error: 'RAND_egd' was not declared in this scope
 RAND_egd(NULL);
              ^
```
Multiple problems here, which I could eventually work around:

- The libressl check ignores `SSL_CFLAGS/CRYPTO_CFLAGS`: fixed in #6729
- `pkg-config --cflags eopenssl` returns `-I/usr/local/include`, should be `-I/usr/local/include/eopenssl` **(upstream issue)**

After applying #6729, explicitly specifying the locations works:
```
./configure ... SSL_CFLAGS="-I/usr/local/include/eopenssl" SSL_LIBS="-L/usr/local/lib/eopenssl -lssl" CRYPTO_CFLAGS="-I/usr/local/include/eopenssl" CRYPTO_LIBS="-L/usr/local/lib/eopenssl -lcrypto"
```

I can forgive you for eventually going with `--with-libressl` flag.

**Comment by dcousens** [comment]
Can be closed?
